src/Features/Core/Portable/InitializeParameter/AbstractAddParameterCheckCodeRefactoringProvider.cs

```diff
@@ -303,6 +317,77 @@ protected bool ParameterValidForNullCheck(Document document, IParameterSymbol pa
         return true;
     }
 
+    private static (IMethodSymbol? GenericOverload, IMethodSymbol? NonGenericOverload) GetEnumIsDefinedMethods(Compilation compilation)
+    {
+        var enumType = compilation.GetSpecialType(SpecialType.System_Enum);
+        var enumIsDefinedMembers = enumType.GetMembers(nameof(Enum.IsDefined));
+        var enumIsDefinedGenericMethod = (IMethodSymbol?)enumIsDefinedMembers.FirstOrDefault(m => m is IMethodSymbol { IsStatic: true, Arity: 1, Parameters.Length: 1 });
+        var enumIsDefinedNonGenericMethod = (IMethodSymbol?)enumIsDefinedMembers.FirstOrDefault(m => m is IMethodSymbol { IsStatic: true, Arity: 0, Parameters.Length: 2 });
+        return (enumIsDefinedGenericMethod, enumIsDefinedNonGenericMethod);
+    }
+
+    private bool ParameterValidForEnumIsDefinedCheck(IParameterSymbol parameter, Compilation compilation, IBlockOperation? blockStatement)
+    {
+        if (parameter.RefKind == RefKind.Out)
+            return false;
+
+        if (parameter.IsDiscard)
+            return false;
+
+        var parameterType = parameter.Type;
+
+        if (parameterType.TypeKind != TypeKind.Enum)
+            return false;
+
+        var flagsAttributeType = compilation.GetBestTypeByMetadataName(typeof(FlagsAttribute).FullName!);
+        if (parameterType.GetAttributes().Any(a => SymbolEqualityComparer.Default.Equals(a.AttributeClass, flagsAttributeType)))
+            return false;
+
+        if (blockStatement is not null)
+        {
+            if (!CanOffer(blockStatement.Syntax))
+                return false;
+
+            var (enumIsDefinedGenericMethod, enumIsDefinedNonGenericMethod) = GetEnumIsDefinedMethods(compilation);
+
+            foreach (var statement in blockStatement.Operations)
+            {
+                if (IsEnumIsDefinedCheck(statement, parameter, enumIsDefinedGenericMethod, enumIsDefinedNonGenericMethod))
+                    return false;
+            }
+        }
+
+        return true;
```

Comment:
why do we default to 'true' if the block is null?

Reply:
The same thing happens in `ParameterValidForNullCheck` method above. IIRC if block body is null it means we have an expression body, which in some cases can be converted to block body later. This logic lies somewhere in the base class for this refactoring